satellite/payments/coinpayments/conversionrates.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package coinpayments
+
+import (
+	"context"
+	"encoding/json"
+	"math/big"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+// cmdRates is API command for retrieving currency rate infos.
+const cmdRates = "rates"
+
+// ExchangeStatus defines if currency is exchangeable.
+type ExchangeStatus string
+
+const (
+	// ExchangeStatusOnline defines exchangeable currency.
+	ExchangeStatusOnline ExchangeStatus = "online"
+	// ExchangeStatusOffline defines currency that can not be convertible at the moment.
+	ExchangeStatusOffline ExchangeStatus = "offline"
+)
+
+// CurrencyRateInfo holds currency conversion info.
+type CurrencyRateInfo struct {
+	IsFiat     bool
+	RateBTC    big.Float
+	TXFee      big.Float
+	Status     ExchangeStatus
+	LastUpdate time.Time
+}
+
+// UnmarshalJSON converts JSON string to currency rate info,
+func (rateInfo *CurrencyRateInfo) UnmarshalJSON(b []byte) error {
+	var rateRaw struct {
+		IsFiat     int    `json:"is_fiat"`
+		RateBTC    string `json:"rate_btc"`
+		TXFee      string `json:"tx_fee"`
+		Status     string `json:"status"`
+		LastUpdate string `json:"last_update"`
+	}
+
+	if err := json.Unmarshal(b, &rateRaw); err != nil {
+		return err
+	}
+
+	const precision uint = 256
```

Comment:
Out of curiosity, why 256 of precision?

Reply:
well because, BTCRate can have more floating points than `transaction` token amount(which has 8), that's to have enough precision to be able to parse more precise values, but not to be dramatic about it